tests/optimisation_test/optimisation_handler.py

```diff
@@ -0,0 +1,48 @@
+import pandas as pd
+import yaml
+from yaml.loader import SafeLoader
+import glob
+from nested_lookup import nested_update
+import os
+
+class RecipeUpdate:
+    def __init__(self, path) -> None:
+        self.files = glob.glob(path)
+        super().__init__()
+
+    def update_values(self, *args):
+        _output_pd = args[0]
+        print(_output_pd)    #TODO remove this
+        _shape = _output_pd.shape
+        _total_columns = _shape[1]
+        with open(self.files[0]) as file:
+            self.data = yaml.load(file, Loader=SafeLoader)
+            print(self.data)
+            for col in range(_total_columns):
+                _key = _output_pd.columns[col]
+                _values = _output_pd[_key].tolist()
+                print(_key,_values)
+                _data_update = nested_update(self.data, key = str(_key), value = _values)
+                self.data = _data_update
```

Comment:
If we add our special annotations as discussed in my comment in the (tests/optimisation_test/algae_bot_recipe_0.yaml) file above. Then you should be able to easily locate where the values need to be inserted and in what format.